src/Control/Applicative/Static.hs

```diff
@@ -0,0 +1,33 @@
+module Control.Applicative.Static where
+
+import Data.Functor.Static
+import Control.Distributed.Closure
+import Data.Typeable (Typeable)
+
+-- | Instances of 'StaticApply' should satisfy the following laws (writing
+-- 'staticMap', 'staticApply' as infix @('<$>')@, @('<*>')@, respectively):
+--
+-- @
+-- static staticCompose '<$>' u '<*>' v '<*>' w = u '<*>' (v '<*>' w)
+-- x '<*>' (f '<$>' y) = (static (flip staticCompose) ``capDup`` f) '<$>' x '<*>' y
+-- f '<$>' (x '<*>' y) = (static staticCompose ``capDup`` f) '<$>' x '<*>' y
+-- @
+--
+-- where
+--
+-- @
+-- staticCompose :: Closure (b -> c) -> Closure (a -> b) -> Closure (a -> c)
+-- staticCompose f g = static (.) ``cap`` f ``cap`` g
```

Comment:
Actually, the instance can't be defined, at least not for `Category` in `base`, because of the `Typeable` constraints. :-/